jwst/extract_1d/extract_1d_step.py

```diff
@@ -186,6 +186,15 @@ def process(self, input):
         elif isinstance(input_model, ModelContainer):
             self.log.debug('Input is a ModelContainer')
         elif isinstance(input_model, datamodels.MultiSlitModel):
+            #  If input is a 3D rateints (which is unsupported) skip the step
+            try:
+                if len((input_model[0]).shape) == 3:
+                    self.log.warning('3D input is unsupported; step will be skipped')
+                    input_model.meta.cal_step.extract_1d = 'SKIPPED'
+                    return input_model
+            except AssertionError:
+                pass
```

Comment:
The same problem with the `try ... except` block here as in resample_step - it's incorrect to check for `AssertionError`. The code already has a check for `MultiSlitModel`. All that's needed is to check that the data array in the first slit is 3D, i.e. `input_model[0].data.shape == 3`

Reply:
Ah, yes I see it now too (mind was numb the first time through). There's no need for a try/except block here. Nothing is going to throw an error that needs to be caught. Simply check for a `MultiSlitModel` as input and then check the shape of that model. If it's 3D, skip.  Same would apply to the test in the resample_spec step.

Reply:
Made those updates to extract_1d_step.py  and resample_spec_step.py